[ERROR] Could not resolve "use-sync-external-store/shim"

I just updated mobx-react-lite from 3.4.3 to 4.0.1 and got this message on the next `npm run dev`:

```console

X [ERROR] Could not resolve "use-sync-external-store/shim"

    node_modules/mobx-react-lite/es/useObserver.js:6:37:
      6 │ import { useSyncExternalStore } from "use-sync-external-store/shim";
        ╵                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "use-sync-external-store/shim" as external to exclude it from the bundle,
  which will remove this error.

C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1636
  let error = new Error(`${text}${summary}`);
              ^

Error: Build failed with 1 error:
node_modules/mobx-react-lite/es/useObserver.js:6:37: ERROR: Could not resolve "use-sync-external-store/shim"
    at failureErrorWithLog (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1636:15)
    at C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1048:25
    at runOnEndCallbacks (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1471:45)
    at buildResponseToResult (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1046:7)
    at C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1058:9
    at new Promise (<anonymous>)
    at requestCallbacks.on-end (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1057:54)
    at handleRequest (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:723:19)
    at handleIncomingPacket (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:745:7)
    at Socket.readFromStdout (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:673:7)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23) {
  errors: [
    {
      detail: undefined,
      id: '',
      location: {
        column: 37,
        file: 'node_modules/mobx-react-lite/es/useObserver.js',
        length: 30,
        line: 6,
        lineText: 'import { useSyncExternalStore } from "use-sync-external-store/shim";',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: null,
          text: 'You can mark the path "use-sync-external-store/shim" as external to exclude it from the bundle, which will remove this error.'
        }
      ],
      pluginName: '',
      text: 'Could not resolve "use-sync-external-store/shim"'
    }
  ],
  warnings: []
}

Node.js v19.4.0
```

Quick check, with what React version is that? And you are importing as ESM,
correct?

On Sun, 26 Mar 2023, 17:14 Alexander Gabriel, ***@***.***>
wrote:

> I just updated mobx-react-lite from 3.4.3 to 4.0.1 and got this message on
> the next npm run dev:
>
> X [ERROR] Could not resolve "use-sync-external-store/shim"
>     node_modules/mobx-react-lite/es/useObserver.js:6:37:      6 │ import { useSyncExternalStore } from "use-sync-external-store/shim";        ╵                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>   You can mark the path "use-sync-external-store/shim" as external to exclude it from the bundle,  which will remove this error.
> C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1636  let error = new Error(`${text}${summary}`);              ^
> Error: Build failed with 1 error:node_modules/mobx-react-lite/es/useObserver.js:6:37: ERROR: Could not resolve "use-sync-external-store/shim"    at failureErrorWithLog (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1636:15)    at C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1048:25    at runOnEndCallbacks (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1471:45)    at buildResponseToResult (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1046:7)    at C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1058:9    at new Promise (<anonymous>)    at requestCallbacks.on-end (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:1057:54)    at handleRequest (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:723:19)    at handleIncomingPacket (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:745:7)    at Socket.readFromStdout (C:\Users\alexa\vermehrung\node_modules\esbuild\lib\main.js:673:7)    at Socket.emit (node:events:513:28)    at addChunk (node:internal/streams/readable:324:12)    at readableAddChunk (node:internal/streams/readable:297:9)    at Readable.push (node:internal/streams/readable:234:10)    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23) {  errors: [    {      detail: undefined,      id: '',      location: {        column: 37,        file: 'node_modules/mobx-react-lite/es/useObserver.js',        length: 30,        line: 6,        lineText: 'import { useSyncExternalStore } from "use-sync-external-store/shim";',        namespace: '',        suggestion: ''      },      notes: [        {          location: null,          text: 'You can mark the path "use-sync-external-store/shim" as external to exclude it from the bundle, which will remove this error.'        }      ],      pluginName: '',      text: 'Could not resolve "use-sync-external-store/shim"'    }  ],  warnings: []}
> Node.js v19.4.0
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mobxjs/mobx/issues/3666>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN4NBDHVVFJ7B3D3KMZUY3W6BTQDANCNFSM6AAAAAAWIHVGQM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

--- a/.changeset/old-candles-wonder.md
+++ b/.changeset/old-candles-wonder.md
@@ -1,0 +1,5 @@
+---
+"mobx-react-lite": patch
+---
+
+Fix package dependency for use-sync-external-store.

--- a/packages/mobx-react-lite/package.json
+++ b/packages/mobx-react-lite/package.json
@@ -36,7 +36,9 @@
         "url": "https://github.com/mobxjs/mobx/issues"
     },
     "homepage": "https://mobx.js.org",
-    "dependencies": {},
+    "dependencies": {
+        "use-sync-external-store": "^1.2.0"
+    },
     "peerDependencies": {
         "mobx": "^6.9.0",
         "react": "^16.8.0 || ^17 || ^18"
@@ -51,8 +53,7 @@
     },
     "devDependencies": {
         "mobx": "^6.9.0",
-        "expose-gc": "^1.0.0",
-        "use-sync-external-store": "^1.2.0"
+        "expose-gc": "^1.0.0"
     },
     "keywords": [
         "mobx",

